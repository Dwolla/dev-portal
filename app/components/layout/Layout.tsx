/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import Head from "next/head";
import { css, Global } from "@emotion/core";
import styled from "@emotion/styled";
import classnames from "classnames";
import { useState, useEffect } from "react";
import Link from "next/link";
import {
  disableBodyScroll,
  enableBodyScroll,
  clearAllBodyScrollLocks,
} from "body-scroll-lock";
import { useRouter } from "next/router";
import SideNav, { SideNavLinkProps } from "./SideNav"; // eslint-disable-line no-unused-vars
import { POPPINS, ROBOTO } from "../typography";
import { BOX_SHADOW_6 } from "../shadowDepths";
import {
  GREY_2,
  GREY_8,
  WHITE_PRIMARY,
  ORANGE_PRIMARY,
  PURPLE_PRIMARY_BUTTON,
  PURPLE_PRIMARY_HOVER,
  PURPLE_PRIMARY_ACTIVE,
  HEADLINE_TEXT,
} from "../colors";
import { breakUp, breakDown } from "../breakpoints";
import TopBar, { TopBarProps, TOP_BAR_HEIGHT } from "./TopBar"; // eslint-disable-line no-unused-vars
import Footer, { FooterLink } from "./Footer"; // eslint-disable-line no-unused-vars
import APIStatusBar from "./APIStatusBar";
import { ReactComponent as CloseIcon } from "../../../assets/images/component-icons/close.svg";
import dwollaDevLogo from "../../../assets/images/dwolla-developers-logo.svg";

export const LEFT_SIDEBAR_PADDING_X = "20px";

const LeftSidebar = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  overflow-y: auto;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  border-right: 1px solid ${GREY_2};
  transition: opacity 300ms;
  opacity: 0;
  pointer-events: none;
  z-index: 9999;

  @media (${breakUp("lg")}) {
    right: 75%;
    opacity: 1;
    pointer-events: auto;
  }

  &.toggled {
    opacity: 1;
    pointer-events: auto;
  }

  /* Hides sidebar on smaller screens when not toggled. Not tabbable when not in view  */
  &.visuallyHidden {
    @media (${breakDown("md")}) {
      visibility: hidden;
    }
  }
`;

const LogoWrapper = styled.div`
  padding: 12px ${LEFT_SIDEBAR_PADDING_X};
  background: white;
  flex-shrink: 1;
  display: flex;
  align-items: center;
  justify-content: space-between;

  @media (${breakDown("md")}) {
    border-bottom: 1px solid ${GREY_2};
    height: ${TOP_BAR_HEIGHT}px;
    min-height: ${TOP_BAR_HEIGHT}px;
  }
`;

const SideNavWrapper = styled.div`
  flex-grow: 1;
  overflow: hidden;
  background: white;
  display: flex;
  flex-direction: column;
`;

const StyledCloseIcon = styled.div`
  cursor: pointer;
  svg {
    * {
      fill: ${ORANGE_PRIMARY};
    }
  }

  :hover,
  &.isFocused {
    svg {
      * {
        fill: ${GREY_8};
      }
    }
  }

  :focus {
    outline: none;
  }

  @media (${breakUp("lg")}) {
    display: none;
  }
`;

const MainArea = styled.div`
  @media (${breakUp("lg")}) {
    margin-left: 25%;
  }
`;

const ContentArea = styled.div`
  @media (${breakUp("lg")}) {
    margin-right: 40px;
  }

  /* Hides ContentArea on smaller screens when SideBar is toggled. Not tabbable when not in view  */
  &.visuallyhidden {
    @media (${breakDown("md")}) {
      visibility: hidden;
    }
  }
`;

const TopBarWrapper = styled.div`
  position: sticky;
  top: 0;
  z-index: 99;
`;

const FooterWrapper = styled.div`
  border-top: 1px solid ${GREY_2};
  background: ${WHITE_PRIMARY};
  z-index: 9;

  @media (${breakUp("xxl")}) {
    padding-right: 40px;
  }

  /* Hides FooterArea on smaller screens when SideBar is toggled. Not tabbable when not in view  */
  &.visuallyhidden {
    @media (${breakDown("md")}) {
      visibility: hidden;
    }
  }
`;

export default function Layout({
  children,
  pages,
  sideNavLinks,
  footerLinks,
  footerLegal,
  topBarProps,
  apiStatus,
}: {
  children: JSX.Element;
  pages: Page[];
  sideNavLinks: SideNavLinkProps[];
  footerLinks: Record<string, FooterLink[]>;
  footerLegal: {
    title: string;
    operatorDescription: string;
    veridianDescription: string;
    metaBankDescription: JSX.Element;
    rtpDescription: string;
  };
  topBarProps: TopBarProps;
  apiStatus: APIStatus;
}) {
  const [sidebarToggled, setSidebarToggled] = useState(false);
  const [isFocused, setIsFocused] = useState(false);

  useEffect(() => {
    if (document) {
      enableBodyScroll(document.querySelector("#body-scroll-lock-side-nav"));
      if (sidebarToggled) {
        disableBodyScroll(document.querySelector("#body-scroll-lock-side-nav"));
      }
    }
    return () => {
      clearAllBodyScrollLocks();
    };
  }, [sidebarToggled]);

  const showSidebar = () => setSidebarToggled(true);
  const hideSidebar = () => setSidebarToggled(false);

  function onKeydown(e) {
    if (e.key === "Enter") hideSidebar();
  }

  const router = useRouter();

  return (
    <>
      <Global
        styles={css`
          * {
            box-sizing: border-box;
          }

          #dwollaCookieConsent {
            font-family: ${ROBOTO};
            color: ${HEADLINE_TEXT};
            line-height: 22px;

            a {
              color: ${ORANGE_PRIMARY};
              text-decoration: none;

              &:hover {
                text-decoration: underline;
              }
            }

            div {
              button {
                font-family: ${POPPINS};
                background-color: ${PURPLE_PRIMARY_BUTTON};
                border-color: transparent;

                :hover,
                :focus {
                  background-color: ${PURPLE_PRIMARY_HOVER};
                  box-shadow: ${BOX_SHADOW_6};
                }

                :active {
                  background-color: ${PURPLE_PRIMARY_ACTIVE};
                }
              }
            }
          }
        `}
      />

      <Head>
        <title>Dwolla Developers</title>
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
        <link
          rel="stylesheet"
          href="https://cdnjs.cloudflare.com/ajax/libs/modern-normalize/0.5.0/modern-normalize.min.css"
        ></link>
        <link
          href="https://fonts.googleapis.com/css?family=Poppins:100,100i,200,200i,300,300i,400,400i,500,500i,600,600i,700,700i,800,800i,900,900i|Roboto:100,100i,300,300i,400,400i,500,500i,700,700i,900,900i&display=swap"
          rel="stylesheet"
        ></link>
        <link rel="canonical" href={router?.pathname} />
        <script
          type="text/javascript"
          src="//js.hsforms.net/forms/v2.js"
        ></script>
      </Head>

      <LeftSidebar
        className={classnames(sidebarToggled ? "toggled" : "visuallyHidden")}
      >
        <LogoWrapper>
          <Link href="/">
            <a>
              <img
                src={dwollaDevLogo}
                alt="Dwolla Developers Logo"
                css={css`
                  @media (${breakDown("md")}) {
                    max-height: 100%;
                  }

                  @media (${breakDown("xs")}) {
                    max-width: 115px;
                  }
                `}
              />
            </a>
          </Link>

          <StyledCloseIcon
            tabIndex={0}
            className={classnames({ isFocused })}
            onClick={hideSidebar}
            onKeyDown={(keyPress) => onKeydown(keyPress)}
            onFocus={() => setIsFocused(true)}
            onBlur={() => setIsFocused(false)}
          >
            <CloseIcon width={14} />
          </StyledCloseIcon>
        </LogoWrapper>

        <SideNavWrapper>
          <SideNav
            sectionLinks={sideNavLinks}
            pages={pages}
            mobileItems={topBarProps}
          />
        </SideNavWrapper>

        <APIStatusBar apiStatus={apiStatus} />
      </LeftSidebar>

      <MainArea>
        <TopBarWrapper>
          <TopBar {...topBarProps} onHamburgerClick={showSidebar} />
        </TopBarWrapper>

        <ContentArea className={classnames({ visuallyhidden: sidebarToggled })}>
          {children}
        </ContentArea>

        <FooterWrapper
          className={classnames({ visuallyhidden: sidebarToggled })}
        >
          <Footer links={footerLinks} legal={footerLegal} />
        </FooterWrapper>
      </MainArea>
    </>
  );
}
