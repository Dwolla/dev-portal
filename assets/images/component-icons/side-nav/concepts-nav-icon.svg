<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 11 17">
  <g fill="#52627B" fill-rule="nonzero">
    <path d="M5.499 0c-.398 0-.795.042-1.185.125a5.491 5.491 0 00-2.88 9.064l.325.351c.902.969 1.24 1.332 1.24 2.46.005.177.058.35.153.5a.977.977 0 00-.153.5v1.5a2.5 2.5 0 005 0V13a.977.977 0 00-.153-.5.977.977 0 00.153-.5c0-1.1.349-1.481 1.192-2.4.134-.146.276-.3.425-.468A5.488 5.488 0 005.499 0zm1.5 14h-3v-1h3v1zm-1.5 2a1.5 1.5 0 01-1.408-1h2.816a1.5 1.5 0 01-1.408 1zm3.37-7.534C7.733 9.742 6.999 10.292 6.999 12h-3c0-1.685-.689-2.244-1.828-3.487A4.491 4.491 0 014.52 1.1c.322-.067.65-.1.979-.1a4.5 4.5 0 014.5 4.5 4.436 4.436 0 01-1.13 2.966z"/>
    <path d="M4.829 2.571a3 3 0 00-2.293 2.448.5.5 0 10.988.151 1.981 1.981 0 011.511-1.62.5.5 0 00-.206-.979z"/>
  </g>
</svg>