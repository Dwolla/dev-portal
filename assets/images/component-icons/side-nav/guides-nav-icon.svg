<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 13 17">
  <g fill="#52627B" fill-rule="evenodd">
    <path d="M12 15a1 1 0 01-1 1H2a1 1 0 01-1-1V3a1 1 0 011-1h2v1h5V2h2a1 1 0 011 1v12zM7.5 1a.5.5 0 01.5.5V2H5v-.5a.5.5 0 01.5-.5h2zM11 1H8.908C8.701.419 8.151 0 7.5 0h-2c-.651 0-1.201.419-1.408 1H2C.897 1 0 1.897 0 3v12c0 1.103.897 2 2 2h9c1.103 0 2-.897 2-2V3c0-1.103-.897-2-2-2z"/>
    <path d="M8.6 7.2l-2.654 3.54-1.593-1.594a.5.5 0 00-.707.707l2 2A.5.5 0 006 12c.012 0 .023 0 .036-.002A.498.498 0 006.4 11.8l3-4a.5.5 0 10-.8-.6"/>
  </g>
</svg>