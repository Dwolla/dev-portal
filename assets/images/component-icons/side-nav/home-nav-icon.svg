<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15 17">
  <path d="M14.6 5.122L8.1.203a1 1 0 00-1.2 0L.4 5.122a1 1 0 00-.4.8v9a2 2 0 002 2h11a2 2 0 002-2v-9a1 1 0 00-.4-.8zM6 15.922v-7h3v7H6zm8-1a1 1 0 01-1 1h-3v-7a1 1 0 00-1-1H6a1 1 0 00-1 1v7H2a1 1 0 01-1-1v-9L7.5 1 14 5.922v9z" fill="#52627B" fill-rule="nonzero"/>
</svg>