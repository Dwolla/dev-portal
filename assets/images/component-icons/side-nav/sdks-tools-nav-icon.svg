<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15 17">
  <path d="M14.5 3.884L8 .134a1 1 0 00-1 0L.5 3.884a1 1 0 00-.5.866v7.5a1 1 0 00.5.866l6.5 3.75a1 1 0 001 0l6.5-3.75a1 1 0 00.5-.866v-7.5a1 1 0 00-.5-.866zM7.5 1l6.029 3.479L7.5 7.924 1.47 4.479 7.5 1zM1 5.362L7 8.79v6.922L1 12.25V5.362zm7 10.35V8.79l6-3.429v6.889l-6 3.462z" fill="#52627B" fill-rule="nonzero"/>
</svg>