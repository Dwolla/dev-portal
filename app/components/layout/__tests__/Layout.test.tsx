import renderer from "react-test-renderer";
import Layout from "../Layout";
import Pages from "../../../modules/pages";

import { ReactComponent as HomeIcon } from "../../../../assets/images/component-icons/side-nav/home-nav-icon.svg";
import { ReactComponent as GuidesIcon } from "../../../../assets/images/component-icons/side-nav/guides-nav-icon.svg";
import { ReactComponent as ConceptsIcon } from "../../../../assets/images/component-icons/side-nav/concepts-nav-icon.svg";
import { ReactComponent as SdksToolsIcon } from "../../../../assets/images/component-icons/side-nav/sdks-tools-nav-icon.svg";

const mockPath = "/";
jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: mockPath,
      pathname: mockPath,
      query: "",
      asPath: mockPath,
    };
  },
}));
jest.mock("../../../modules/images.import");

const SIDE_NAV_LINKS = [
  {
    href: "/",
    IconSvg: HomeIcon,
    isSection: false,
    text: "Home",
  },
  {
    href: "/guides",
    IconSvg: GuidesIcon,
    isSection: true,
    text: "Guides",
  },
  {
    href: "/concepts",
    IconSvg: ConceptsIcon,
    isSection: true,
    text: "Concepts",
  },
  {
    href: "/sdks-tools",
    IconSvg: SdksToolsIcon,
    isSection: false,
    text: "SDKs & Tools",
  },
];

const FOOTER_LINKS = {
  Dwolla: [
    { text: "About", href: "https://www.dwolla.com/about/" },
    { text: "Blog", href: "https://www.dwolla.com/updates/" },
    { text: "Pricing", href: "https://www.dwolla.com/pricing/" },
    { text: "Contact Sales", href: "https://www.dwolla.com/contact/" },
    { text: "Terms of Service", href: "https://www.dwolla.com/legal/tos/" },
    { text: "Privacy Policy", href: "https://www.dwolla.com/legal/privacy/" },
  ],
  Product: [
    { text: "Resources", href: "https://www.dwolla.com/resources/" },
    { text: "Features", href: "https://www.dwolla.com/features/" },
    { text: "Case Studies", href: "https://www.dwolla.com/case-studies/" },
    { text: "Integrations", href: "https://www.dwolla.com/integrations/" },
  ],
  Developers: [
    { text: "API Reference", href: "https://docs.dwolla.com/" },
    { text: "Support Forum", href: "https://discuss.dwolla.com/" },
    { text: "SDKs and Tools", href: "#sdks-and-tools" },
    { text: "Github Repos", href: "https://github.com/Dwolla/" },
  ],
};

const FOOTER_LEGAL_COPY = {
  title: "Financial institutions play an important role in our network.",
  veridianDescription:
    "Dwolla, Inc. is an agent of Veridian Credit Union and all funds associated with your account in our network are held in one or more pooled accounts at Veridian Credit Union. These funds may not be eligible for share insurance by the National Credit Union Share Insurance Fund. Dwolla, Inc. is the operator of a software platform that communicates user instructions for funds transfers to Veridian Credit Union.",
  metaBankDescription: (
    <span>
      Sponsorship and Settlement of Push-to-Debit payment services provided by{" "}
      <span style={{ whiteSpace: "nowrap" }}>MetaBankÂ®, N.A.</span>
      <br /> Push-to-Debit payments are typically available within 30 minutes.
    </span>
  ),
};

const TOP_BAR_PROPS = {
  button: {
    text: "Get API Keys",
    link: {
      href: "https://accounts-sandbox.dwolla.com/login",
      external: true,
    },
  },
  links: [
    { text: "API Docs", href: "https://docs.dwolla.com", external: true },
    { text: "Changelog", href: "/changelog" },
  ],
};

test("Layout", () => {
  const tree = renderer
    .create(
      <Layout
        sideNavLinks={SIDE_NAV_LINKS}
        footerLinks={FOOTER_LINKS}
        footerLegal={FOOTER_LEGAL_COPY}
        topBarProps={TOP_BAR_PROPS}
        pages={Pages.all()}
        apiStatus={{ indicator: "none", description: "apiStatus description" }}
      >
        <div>page</div>
      </Layout>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
