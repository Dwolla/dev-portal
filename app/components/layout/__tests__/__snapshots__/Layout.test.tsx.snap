// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout 1`] = `
[
  <div
    className="css-1h0fusp-TopBarWrapper e1ijxxr31"
  >
    <div
      className="css-y5wall-Container e1psldpd6"
    >
      <a
        href="/docs"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "textDecoration": "none",
          }
        }
      >
        <img
          alt=""
          className="css-3jqt57-StyledLogo e1psldpd5"
          src="test-image-stub"
        />
        <em
          className="css-11c4vbk-StyledEm e1psldpd4"
        >
           Docs
        </em>
      </a>
      <div
        className="css-5vssss-LeftAlignWrapper e1psldpd2"
      />
      <div
        className="css-j4xy9d-RightAlignWrapper e1psldpd3"
      >
        <div
          className="MuiFormControl-root css-dru9rs-MuiFormControl-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            id=":r0:"
          >
            Select Language
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            onClick={[Function]}
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby=":r0:"
              className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="notranslate"
              >
                â€‹
              </span>
            </div>
            <input
              aria-hidden={true}
              aria-invalid={false}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value=""
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Select Language
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <a
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
          href="https://accounts-sandbox.dwolla.com/login"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          target="_blank"
        >
          Get API Keys
        </a>
      </div>
      <div
        aria-labelledby="Open menu"
        className="css-mj3r16-Hamburger e1psldpd1"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
      />
    </div>
  </div>,
  <div
    className="css-1lvinfx-LayoutContainer e1ijxxr36"
  >
    <div
      className="visuallyHidden css-kjw9mq-LeftSidebar e1ijxxr35"
    >
      <div
        className="css-6qgefa-SideNavWrapper e1ijxxr34"
      >
        <div
          className="css-yeohq5-Container e1qiqw1s17"
        >
          <div
            className="css-kibqvq-Slide e1qiqw1s16"
          >
            <div
              className="left css-soqej8-SlidePane e1qiqw1s15"
              id="body-scroll-lock-side-nav"
            >
              <div
                className="css-kgvw59-SectionWrap e1qiqw1s14"
              >
                <a
                  className=" css-nr9jvx-SectionLink-SectionLink"
                  href="/docs"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onTouchStart={[Function]}
                >
                  <div
                    className="css-30rc4k-SectionIconWrap e1qiqw1s11"
                  >
                    <svg
                      className="section-icon"
                      height={22}
                      width={22}
                    />
                  </div>
                  Home
                </a>
                <a
                  className=" css-nr9jvx-SectionLink-SectionLink"
                  href="/docs/balance"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onTouchStart={[Function]}
                >
                  <div
                    className="css-30rc4k-SectionIconWrap e1qiqw1s11"
                  >
                    <svg
                      className="section-icon"
                      height={22}
                      width={22}
                    />
                  </div>
                  Balance
                </a>
                <a
                  className=" css-nr9jvx-SectionLink-SectionLink"
                  href="/api-reference"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onTouchStart={[Function]}
                >
                  <div
                    className="css-30rc4k-SectionIconWrap e1qiqw1s11"
                  >
                    <svg
                      className="section-icon"
                      height={22}
                      width={22}
                    />
                  </div>
                  API Reference
                  <div
                    className="css-1qriqu9-SectionCaretWrap e1qiqw1s10"
                  >
                    <svg
                      height={18}
                      width={18}
                    />
                  </div>
                </a>
                <a
                  className=" css-nr9jvx-SectionLink-SectionLink"
                  href="/sdks-tools"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onTouchStart={[Function]}
                >
                  <div
                    className="css-30rc4k-SectionIconWrap e1qiqw1s11"
                  >
                    <svg
                      className="section-icon"
                      height={22}
                      width={22}
                    />
                  </div>
                  SDKs & Tools
                </a>
              </div>
              <div
                className="css-17mreui-ExternalSectionWrap e1qiqw1s13"
              >
                <a
                  className=" css-nr9jvx-SectionLink-SectionLink"
                  href="https://discuss.dwolla.com/"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onTouchStart={[Function]}
                  target="_blank"
                >
                  <div
                    className="css-30rc4k-SectionIconWrap e1qiqw1s11"
                  >
                    <svg
                      className="section-icon"
                      height={15}
                      width={15}
                    />
                  </div>
                  Community
                </a>
              </div>
              <div
                className="css-19j3ofi-MobileWrap e1qiqw1s1"
              >
                <div
                  className="css-1rc904k-MobileButtonWrap e1qiqw1s0"
                >
                  <a
                    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-mrzqee-MuiButtonBase-root-MuiButton-root"
                    href="https://accounts-sandbox.dwolla.com/login"
                    onBlur={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    target="_blank"
                  >
                    Get API Keys
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <a
        className="css-pnlj05-ContainerLink eyy6ry91"
        href="https://status.dwolla.com"
        target="_blank"
      >
        <span
          className="indicator-none css-ulyjnq-StatusIcon eyy6ry90"
        />
        apiStatus description
      </a>
    </div>
    <div
      className="css-1v2y2k5-MainArea e1ijxxr33"
    >
      <div
        className=" css-x5ql3q-ContentArea e1ijxxr32"
      >
        <div>
          page
        </div>
      </div>
      <div
        className=" css-17pvfhk-FooterWrapper e1ijxxr30"
      >
        <div
          className="css-rntgrl-Container eg0edpy12"
        >
          <div
            className="css-549mvs-LogoContainer eg0edpy11"
          >
            <a
              href="/docs"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              <img
                alt=""
                className="css-1289bd9-Logo eg0edpy10"
                src="test-image-stub"
              />
            </a>
            <div
              className="css-1neb9ej-Copyright eg0edpy9"
            >
              2023
               All Rights Reserved
            </div>
          </div>
          <div
            className="css-1g5yeab-LinkContainer eg0edpy8"
          >
            <div
              className="css-n2kuza-LinkGroup eg0edpy7"
            >
              <h3
                className="css-ve8wjm-LinkGroupHeading eg0edpy6"
              >
                Dwolla
              </h3>
              <ul
                className="css-1wxmwh7-LinkGroupList eg0edpy5"
              >
                <li
                  className="css-146nff3-LinkGroupItem eg0edpy4"
                >
                  <a
                    className="css-1yfy9it-LinkGroupLink eg0edpy3"
                    href="https://www.dwolla.com/about/"
                  >
                    About
                  </a>
                </li>
                <li
                  className="css-146nff3-LinkGroupItem eg0edpy4"
                >
                  <a
                    className="css-1yfy9it-LinkGroupLink eg0edpy3"
                    href="https://www.dwolla.com/updates/"
                  >
                    Blog
                  </a>
                </li>
                <li
                  className="css-146nff3-LinkGroupItem eg0edpy4"
                >
                  <a
                    className="css-1yfy9it-LinkGroupLink eg0edpy3"
                    href="https://www.dwolla.com/pricing/"
                  >
                    Pricing
                  </a>
                </li>
                <li
                  className="css-146nff3-LinkGroupItem eg0edpy4"
                >
                  <a
                    className="css-1yfy9it-LinkGroupLink eg0edpy3"
                    href="https://www.dwolla.com/contact/"
                  >
                    Contact Sales
                  </a>
                </li>
                <li
                  className="css-146nff3-LinkGroupItem eg0edpy4"
                >
                  <a
                    className="css-1yfy9it-LinkGroupLink eg0edpy3"
                    href="https://www.dwolla.com/legal/tos/"
                  >
                    Terms of Service
                  </a>
                </li>
                <li
                  className="css-146nff3-LinkGroupItem eg0edpy4"
                >
                  <a
                    className="css-1yfy9it-LinkGroupLink eg0edpy3"
                    href="https://www.dwolla.com/legal/privacy/"
                  >
                    Privacy Policy
                  </a>
                </li>
              </ul>
            </div>
            <div
              className="css-n2kuza-LinkGroup eg0edpy7"
            >
              <h3
                className="css-ve8wjm-LinkGroupHeading eg0edpy6"
              >
                Product
              </h3>
              <ul
                className="css-1wxmwh7-LinkGroupList eg0edpy5"
              >
                <li
                  className="css-146nff3-LinkGroupItem eg0edpy4"
                >
                  <a
                    className="css-1yfy9it-LinkGroupLink eg0edpy3"
                    href="https://support.dwolla.com/s/"
                  >
                    Support
                  </a>
                </li>
                <li
                  className="css-146nff3-LinkGroupItem eg0edpy4"
                >
                  <a
                    className="css-1yfy9it-LinkGroupLink eg0edpy3"
                    href="https://www.dwolla.com/resources/"
                  >
                    Resources
                  </a>
                </li>
                <li
                  className="css-146nff3-LinkGroupItem eg0edpy4"
                >
                  <a
                    className="css-1yfy9it-LinkGroupLink eg0edpy3"
                    href="https://www.dwolla.com/features/"
                  >
                    Features
                  </a>
                </li>
                <li
                  className="css-146nff3-LinkGroupItem eg0edpy4"
                >
                  <a
                    className="css-1yfy9it-LinkGroupLink eg0edpy3"
                    href="https://www.dwolla.com/case-studies/"
                  >
                    Case Studies
                  </a>
                </li>
              </ul>
            </div>
            <div
              className="css-n2kuza-LinkGroup eg0edpy7"
            >
              <h3
                className="css-ve8wjm-LinkGroupHeading eg0edpy6"
              >
                Developers
              </h3>
              <ul
                className="css-1wxmwh7-LinkGroupList eg0edpy5"
              >
                <li
                  className="css-146nff3-LinkGroupItem eg0edpy4"
                >
                  <a
                    className="css-1yfy9it-LinkGroupLink eg0edpy3"
                    href="https://developers.dwolla.com/api-reference/"
                  >
                    API Reference
                  </a>
                </li>
                <li
                  className="css-146nff3-LinkGroupItem eg0edpy4"
                >
                  <a
                    className="css-1yfy9it-LinkGroupLink eg0edpy3"
                    href="https://discuss.dwolla.com/"
                  >
                    Support Forum
                  </a>
                </li>
                <li
                  className="css-146nff3-LinkGroupItem eg0edpy4"
                >
                  <a
                    className="css-1yfy9it-LinkGroupLink eg0edpy3"
                    href="#sdks-and-tools"
                  >
                    SDKs and Tools
                  </a>
                </li>
                <li
                  className="css-146nff3-LinkGroupItem eg0edpy4"
                >
                  <a
                    className="css-1yfy9it-LinkGroupLink eg0edpy3"
                    href="https://github.com/Dwolla/"
                  >
                    Github Repos
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="css-1kena77-LegalContainer eg0edpy2"
          >
            <strong
              className="css-31ngrj-LegalTitle eg0edpy1"
            >
              Financial institutions play an important role in our network.
            </strong>
            <p
              className="css-b6puy4-LegalDesc eg0edpy0"
            >
              All funds transfers made using the Dwolla Platform are performed by a financial institution partner, and any funds held in a Dwolla Balance are held by a financial institution partner.
               
              <a
                href="https://www.dwolla.com/legal/about-our-financial-institution-partners/#legal-content"
                rel="noreferrer"
                target="_blank"
              >
                Learn more about our financial institution partners
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
